shub_image/list.py

```diff
@@ -0,0 +1,123 @@
+"""
+
+
+@click.command(help=HELP, short_help=SHORT_HELP)
+@click.argument("target", required=False, default="default")
+@click.option("-l", "--list-targets", help="list available targets",
+              is_flag=True, is_eager=True, expose_value=False,
+              callback=list_targets)
+@click.option("-d", "--debug", help="debug mode", is_flag=True)
+@click.option("-s", "--silent", help="silent mode", is_flag=True)
+@click.option("--version", help="release version")
+@click.pass_context
+def cli(ctx, target, debug, silent, version):
+    ctx.obj = {'debug': debug}
+    list_cmd_full(target, silent, version)
+
+
+def list_cmd_full(target, silent, version):
+    config = utils.load_release_config()
+    image = config.get_image(target)
+    version = version or config.get_version()
+    image_name = utils.format_image_name(image, version)
+    project, endpoint, apikey = None, None, None
+    try:
+        project, endpoint, apikey = config.get_target(target)
+    except shub_exceptions.BadParameterException as exc:
+        if 'Could not find target' not in exc.message:
+            raise
+        if not silent:
+            click.echo(
+                "Not found project for target {}, "
+                "not getting project settings from Dash.".format(target))
+    spiders = list_cmd(image_name, project, endpoint, apikey)
+    for spider in spiders:
+        click.echo(spider)
+
+
+def list_cmd(image_name, project, endpoint, apikey):
+    """Short version of list cmd to use with deploy cmd."""
+
+    settings = {}
+    if project:
+        settings = _get_project_settings(project, endpoint, apikey)
+
+    # Run a local docker container to run list-spiders cmd
+    status_code, logs = _run_list_cmd(project, image_name, settings)
+    if status_code != 0:
+        click.echo(logs)
+        raise shub_exceptions.ShubException(
+            'Container with list cmd exited with code %s' % status_code)
+
+    spiders = utils.valid_spiders(logs)
+    return spiders
+
+
+def _get_project_settings(project, endpoint, apikey):
+    utils.debug_log('Getting settings for {} project:'.format(project))
+    req = requests.get(
+        urljoin(endpoint, '/api/settings/get.json'),
+        params={'project': project},
+        auth=(apikey, ''),
+        timeout=300,
+        allow_redirects=False
+    )
+    req.raise_for_status()
+    utils.debug_log("Response: {}".format(req.json()))
+    return {k: v for k, v in req.json().items() if k in SETTING_TYPES}
+
+
+def _run_list_cmd(project, image_name, project_settings):
+    """Run `scrapy list` command inside the image container."""
+
+    client = utils.get_docker_client()
+    # FIXME we should pass some value for SCRAPY_PROJECT_ID anyway
+    # to handle `scrapy list` cmd properly via sh_scrapy entrypoint
+    project = str(project) if project else ''
+    job_settings = json.dumps(project_settings)
+    container = client.create_container(
+        image=image_name,
+        command=['list-spiders'],
```

Comment:
As we rely on latest `scrapinghub-entrypoint-scrapy=0.7.0` with this change and using new `list-spiders` entrypoint instead of running `scrapy list` locally, probably it makes sense to update `README` as well?